### DISCUSSION TITLE
Release Plz Seems to Error When Rustup Has to Download a Toolchain in The Middle of Updates

### DISCUSSION THREAD

**Discussion by zicklag** [release-plz/release-plz#1203] [Q&A] [answered]
Hey there! I'm really liking this project, but I'm having a little bit of trouble.

When release please does the package comparisons during a `release-plz release-pr` run, it downloads the old versions of my crates, which contain a `rust-toolchain` file. This causes rustup to download and install the version of rust specified in the old crates `rust-toolchain` file, and `release-plz` will error out:

```
2024-01-14T17:41:46.445944Z ERROR  cannot compare packages

Caused by:
    0: cannot determine packaged files of local package "/tmp/.tmp5ajgZP/bones/framework_crates/bones_asset"
    1: error while running `cargo package`: info: syncing channel updates for '1.73-x86_64-unknown-linux-gnu'
       info: latest update on [202](https://github.com/fishfolk/bones/actions/runs/7520769460/job/20470792915#step:4:207)3-10-05, rust version 1.73.0 (cc66ad468 2023-10-03)
       info: downloading component 'cargo'
       info: downloading component 'rust-std'
       info: downloading component 'rustc'
       info: installing component 'cargo'
       info: installing component 'rust-std'
       info: installing component 'rustc'
```

For example on this run: https://github.com/fishfolk/bones/actions/runs/7520769460/job/20470792915#step:4:204

If I install the proper version of Rust ahead of time, it works fine, because rustup doesn't output anything extra maybe?

I didn't open a bug since I don't have fully reproduction steps yet, but I wanted to open the discussion just to make anybody else aware who might run into this.

I think the only workaround right now would be to set a rustup override for the `/tmp` folder, but I haven't tried it yet.

**Comment by zicklag** [comment]
> I think the only workaround right now would be to set a rustup override for the /tmp folder, but I haven't tried it yet.

Ah, bummer, doing `rustup override set --path /tmp stable` didn't work. :/ probably I have to set overrides for each crate.

I'd just install the Rust version I need ahead of time if it weren't for the fact that the Rust version will change across crate releases, because we basically follow Rust stable, I just set the toolchain version specifically so it doesn't randomly change the Rust version.

**Comment by marcoieni** [comment]
we can fix this by changing [this](https://github.com/MarcoIeni/release-plz/blob/ab9432c67881c8185c670ca3d355a61fefc8f922/crates/release_plz_core/src/package_compare.rs#L104) check.
Right now, we error if stderr is not empty. The fix can be to check if the error starts with "info: syncing channel updates for".

Do you want to submit a pr? :)

**Reply by zicklag** [reply]
Thanks for tracing it. I'll try to submit a PR as soon as I get back to trying to finish up my releases. 👍️

**Reply by zicklag** [reply]
@MarcoIeni would it make more sense to check whether or not there is a non-zero exit code instead of whether or not there is any stderr output? It seems like it might not be a good idea to depend on the output of rustup which could feasibly change between releases.

**Reply by marcoieni** [reply]
Yes, I think if makes more sense 👍

**Comment by marcoieni** [comment, marked as answer]
merged the fix: https://github.com/MarcoIeni/release-plz/pull/1234

**Reply by zicklag** [reply]
Thank you! ❤️
